app/src/main/java/org/simple/clinic/editpatient/OngoingEditPatientEntry.kt

```diff
@@ -112,47 +117,65 @@ data class OngoingEditPatientEntry @Deprecated("Use the `from` factory function
       numberValidator: PhoneNumberValidator,
       dobValidator: UserInputDateValidator
   ): Set<EditPatientValidationError> {
-    val errors = mutableSetOf<EditPatientValidationError>()
+    return getValidationChecks(alreadySavedNumber, numberValidator, ageOrDateOfBirth, dobValidator)
+        .mapNotNull { check -> check() }
+        .toSet()
+  }
 
-    if (name.isBlank()) {
-      errors.add(FULL_NAME_EMPTY)
-    }
+  private fun getValidationChecks(
+      alreadySavedNumber: PatientPhoneNumber?,
+      numberValidator: PhoneNumberValidator,
+      ageOrDateOfBirth: EitherAgeOrDateOfBirth,
+      dobValidator: UserInputDateValidator
+  ): List<ValidationCheck> {
+    return listOf(
+        nameCheck(),
+        phoneNumberCheck(alreadySavedNumber, numberValidator),
+        colonyOrVillageCheck(),
+        stateCheck(),
+        districtCheck(),
+        ageOrDateOfBirthCheck(ageOrDateOfBirth, dobValidator)
+    )
+  }
 
-    when (numberValidator.validate(phoneNumber, Type.LANDLINE_OR_MOBILE)) {
-      Result.LENGTH_TOO_SHORT -> errors.add(PHONE_NUMBER_LENGTH_TOO_SHORT)
-      Result.LENGTH_TOO_LONG -> errors.add(PHONE_NUMBER_LENGTH_TOO_LONG)
-      Result.BLANK -> alreadySavedNumber?.let { errors.add(PHONE_NUMBER_EMPTY) }
+  private fun nameCheck(): ValidationCheck =
+      { if (name.isBlank()) FULL_NAME_EMPTY else null }
 
-      Result.VALID -> {
-        // Do nothing here
-      }
-    }
-
-    if (colonyOrVillage.isBlank()) {
-      errors.add(COLONY_OR_VILLAGE_EMPTY)
+  private fun phoneNumberCheck(
+      alreadySavedNumber: PatientPhoneNumber?,
+      numberValidator: PhoneNumberValidator
+  ): ValidationCheck = {
+    when (numberValidator.validate(phoneNumber, Type.LANDLINE_OR_MOBILE)) {
+      LENGTH_TOO_SHORT -> PHONE_NUMBER_LENGTH_TOO_SHORT
+      LENGTH_TOO_LONG -> PHONE_NUMBER_LENGTH_TOO_LONG
+      BLANK -> if (alreadySavedNumber != null) PHONE_NUMBER_EMPTY else null
+      VALID -> null
     }
+  }
 
-    if (state.isBlank()) {
-      errors.add(STATE_EMPTY)
-    }
+  private fun colonyOrVillageCheck(): ValidationCheck =
+      { if (colonyOrVillage.isBlank()) COLONY_OR_VILLAGE_EMPTY else null }
 
-    if (district.isBlank()) {
-      errors.add(DISTRICT_EMPTY)
-    }
+  private fun stateCheck(): ValidationCheck =
+      { if (state.isBlank()) STATE_EMPTY else null }
 
-    if (ageOrDateOfBirth is EntryWithDateOfBirth) {
-      when (dobValidator.validate(ageOrDateOfBirth.dateOfBirth)) {
-        InvalidPattern -> errors.add(INVALID_DATE_OF_BIRTH)
-        DateIsInFuture -> errors.add(DATE_OF_BIRTH_IN_FUTURE)
-        is Valid -> { /* Nothing to do here. */ }
-      }
+  private fun districtCheck(): ValidationCheck =
+      { if (district.isBlank()) DISTRICT_EMPTY else null }
 
-    } else if (ageOrDateOfBirth is EntryWithAge) {
-      if (ageOrDateOfBirth.age.isBlank()) {
-        errors.add(BOTH_DATEOFBIRTH_AND_AGE_ABSENT)
+  private fun ageOrDateOfBirthCheck(
+      ageOrDateOfBirth: EitherAgeOrDateOfBirth,
```

Comment:
This parameter is unnecessary since this is already a member of `OngoingEditPatientEntry`.